### PR TITLE
adds a number of naming convention guides

### PR DESCRIPTION
**Pull request by kilbergr**
Eady and I are working with the content guild on expanding infrasec learning and style resources.
We spent time going through some of the resources and services we often encounter and struggle with naming over. We attempted to stay aligned with what's written here as well as what's historically been the status quo. We got too loopy after about 45 minutes so this is as far as we got.
This is the ideas board we're working from: https://miro.com/app/board/o9J_l4KAX5c=/

We'll plan to come back and continue fleshing this out.

### CODE REVIEW

File: infrasec/aws/naming.md [hunk 1]
```diff
@@ -79,3 +91,54 @@ e.g.
...
+* *$environment* - can be used to distinguish different versions of the resource/app that occur during the development lifecycle, e.g. `dev`, `perf_test`, `staging` and `prod`uction
+
+* eec-prod
+* staging
+
+### Subnets
+
+Although they are unique by region, we also want subnets to be descriptively named according to their AZs and public/private status.
+
+*${VPC name}-${public/private}-${AZ}* - is the general form
```
Comment (cblkwell) [inline infrasec/aws/naming.md:122]: Technically there are also "db" subnets that could be defined as part of a VPC.

File: infrasec/aws/naming.md [hunk 2]
```diff
@@ -79,3 +91,54 @@ e.g.
...
+
+* *$public/private* - can be used to distinguish whether the subnet is a private or a public subnet
+* *$AZ* - Availability Zone (AZ) are within a region and are ordered initials (i.e. `us-east-2b`).
+
+In a VPC named "eec-prod":
+
+* eec-prod-public-us-east-1a
+* eec-prod-private-us-west-2c
+
+## Terraform modules
```
Comment (cblkwell) [inline infrasec/aws/naming.md:132]: Might be worth noting here that this is the format the registry *requires* in order for your module to be properly added to it, so this is less of a guideline and more of a rule.

**Review by cblkwell — CHANGES_REQUESTED** [review]
Looks pretty good, just two additional notes.

File: infrasec/aws/naming.md (hunk 2 above)
Comment (kilbergr) [reply infrasec/aws/naming.md:132]: Ohhh man. We thought that might be but then I thought I found a counterexample I can't find again so thank you!

**Comment by kilbergr** [comment]
@rpdelaney we should. I will try to pick it up. Given everyone who commented on this is no longer here...can I tag you to review?

**Comment by kilbergr** [comment]
@rpdelaney ready for review!

**Review by rpdelaney — APPROVED** [review]

File: infrasec/aws/naming.md [hunk 3]
```diff
@@ -77,3 +92,89 @@ e.g.
 
 - slack-pivotal-tracker-bot-test
 - rds-log-cleaner-webapp-prod
+
+### SNS topics
+
+The main purpose of the SNS topics are for notifications to teams. The naming convention reflects that, closely followed by the AWS account type (Gov or Com) and action.
+
+*${team}-${account_type}-${action}* - is the general form
```
Comment (rpdelaney) [inline infrasec/aws/naming.md:100]:
```suggestion
`${team}-${account_type}-${action}` - is the general form
```
Not blocking: subjectively, I find this a bit easier to read than italics. There are a few more instances later on also, if you agree. If not, it's no kind of big deal :)

**Pull request merged**